### DISCUSSION TITLE
Columns

### DISCUSSION THREAD

**Discussion by marijang** [infinum/eightshift-frontend-libs#718] [Q&A] [answered]
I have problem with columns and 12 / 14 column spec

Scenario:

First is component HEADING: wrapper offset 1
Second is columns: 
 1. column (width:4:offset 2) 

To get in same position i need to play with column gap....i need to put 50...
Also when i use that scenario then remains only 10 columns which is not divided by three:(

**Comment by goranalkovic-infinum** [comment]
To preface, you should avoid nesting Columns blocks inside other Columns block, as it adds unnecessary HTML elements.

Regarding the number of columns - in the global manifest you can set the total number of the columns in your project, so 13 might work better for you.

If it's a very specific layout, making a custom block could also be a viable solution.

**Reply by marijang** [reply]
i m not using nesting columns

**Reply by goranalkovic-infinum** [reply]
Ah, ok, so it's a heading block, below which is a Columns block?

**Reply by marijang** [reply]
Yes.

**Comment by marijang** [comment]
wait i have link, you can see below services... three boxes blue

https://cop.wp.onlyoneif.com/usluge/

**Comment by goranalkovic-infinum** [comment]
In that case, to ensure perfect alignment, you could place the heading also inside the same Columns block, and have its containing Column span full width.

Then, all the columns you add below will just wrap to the lines below it.

**Comment by marijang** [comment]
a yea...to much 
or just put 12 column mode on columns:D

**Comment by marijang** [comment]
But you see now example of use.

**Comment by goranalkovic-infinum** [comment, marked as answer]
Yup, the example is clear now.

With slight modifications within the Columns styles and options you can really easily achieve columns without gutters ("12-column columns") :)

**Reply by marijang** [reply]
how?

**Reply by marijang** [reply]
or just enable on columns to have full size and 12 columns....like other wrappers:D

**Comment by marijang** [comment]
i want to use 14 and 12 columns option. Why to have restriction?
I m not fan of modification your standard components because update.

**Reply by goranalkovic-infinum** [reply]
There are no restrictions, we just didn't need the option so it wasn't implemented :)
The _Columns_ block can't have a wrapper, so the only logical outcome for use was to have all the columns available to place items in.

Regarding modifying "the built-in components", you can (and are encouraged to) do that inside your project, as (unlike with Core WP) updating Frontend Libs won't affect the blocks, components and wrapper that are in your project.

P.S.: feel free to make a PR if you end up implementing the Full-width toggle for Columns and you want to share it with everyone 😉

**Reply by marijang** [reply]
is it enaught to put on columns

"wrapperUseShowControl":  {
	"type": "boolean",
	"default": true
},

Then i will get 12 columns

**Reply by marijang** [reply]
and on column i need to put toggle 12 columns vs 14 and also css fix

**Reply by goranalkovic-infinum** [reply]
No, you need to:
1. remove the columns from the `columns-style.scss`
2. change the column picker to 12-column mode and disable the gutters in `column-options.js`

and then if you want to enable the full wrapper on Column**s**, you can.

**Reply by marijang** [reply]
yes but i need to pass 12 columns to column block. How to achieve that

**Reply by goranalkovic-infinum** [reply]
<img width="686" alt="image" src="https://user-images.githubusercontent.com/77000136/235670007-1f28c119-3ad2-4b3a-9830-a4a708e97e77.png">

**Comment by marijang** [comment]
In generateWidthOffsetRangeSliderConfig you don't have includeGutters

**Reply by marijang** [reply]
also when i put in columns-style.scss
--columns-grid-side-columns: calc(var(--global-grid-side-padding) - calc(var(--columns-column-gap, 0) * var(--base-font-size) * 0.1rem));

then i get nice offset in 14 columns...because now is aligned
<img width="307" alt="Screen Shot 2023-05-04 at 11 53 11" src="https://user-images.githubusercontent.com/1381443/236171803-08d8f713-96fd-4701-a355-233c39a2b030.png">

**Reply by goranalkovic-infinum** [reply]
> In generateWidthOffsetRangeSliderConfig you don't have includeGutters

Sorry, could you elaborate this a bit more? The config option is definitely there 😄 Perhaps you're looking at _Column**s**_ instead of _Column_?

> also when i put in columns-style.scss
--columns-grid-side-columns: calc(var(--global-grid-side-padding) - calc(var(--columns-column-gap, 0) * var(--base-font-size) * 0.1rem));
> 
> then i get nice offset in 14 columns...because now is aligned

That's one of the solutions for sure, you could also add some inline padding to the Column**s** block.

**Comment by marijang** [comment]
look
<img width="242" alt="Screen Shot 2023-05-04 at 11 58 39" src="https://user-images.githubusercontent.com/1381443/236173313-ec31cad6-d748-4f09-8560-0bb5de0d1770.png">

**Reply by marijang** [reply]
<img width="1328" alt="Screen Shot 2023-05-04 at 12 38 42" src="https://user-images.githubusercontent.com/1381443/236181239-017058ba-5b36-4a61-93b7-8e4964ae84a7.png">
basicly my problem is not 12 columns...
if i want to use offset 2 columns left and right and divide rest of the content in 3 parts i cant do that because i have only 10 columns...10 I can't divide in 3 columns same sizes

**Reply by goranalkovic-infinum** [reply]
You're not locked into 12 columns, it's always possible to have as many as you want, but in that case if you have 12 columns in the rest of the site, you might not be able to align things well if Columns are 13 columns for example.

**Reply by goranalkovic-infinum** [reply]
To change the number of columns in the Columns block, simply change the `grid-template-columns` to have 13 (or any other number) of columns instead of the default `var(--global-main-content-width)`.

After that, you need to change the `numOfColumns` in the _Column_ block, inside `generateWidthOffsetRangeSliderConfig`.

**Comment by iruzevic** [comment]
Thank you, @marijang, for your invaluable input on this ticket. However, as Goran mentioned, this is just a boilerplate. Once you've copied the blocks into your project, you can edit, update, delete, or add anything you'd like. So, there's no need to complicate things any further. 

From the conversation, it seems like there are no bugs on this ticket, just suggestions. Goran confirmed that everything is okay with the block, so I will be closing this issue. 

Feel free to open a PR or make a fork of this setup and create the boilerplate that suits your needs.
